store/localstore/local_region.go

```diff
@@ -44,18 +46,131 @@ type regionResponse struct {
 
 const chunkSize = 64
 
+type sortRow struct {
+	key  []types.Datum
+	meta tipb.RowMeta
+	data []byte
+}
+
+type topnContainer struct {
+	orderByItems []*tipb.ByItem
+	rows         []*sortRow
+	err          error
+}
+
+func (t *topnContainer) Len() int {
+	return len(t.rows)
+}
+
+func (t *topnContainer) Swap(i, j int) {
+	t.rows[i], t.rows[j] = t.rows[j], t.rows[i]
+}
+
+func (t *topnContainer) Less(i, j int) bool {
+	for index, by := range t.orderByItems {
+		v1 := t.rows[i].key[index]
+		v2 := t.rows[j].key[index]
+
+		ret, err := v1.CompareDatum(v2)
+		if err != nil {
+			t.err = errors.Trace(err)
+			return true
+		}
+
+		if by.Desc {
+			ret = -ret
+		}
+
+		if ret < 0 {
+			return true
+		} else if ret > 0 {
+			return false
+		}
+	}
+
+	return false
+}
+
+type topnSolver struct {
+	topnContainer
+
+	totalCount int
+	heapSize   int
+}
+
+func (t *topnSolver) Len() int {
+	return t.heapSize
+}
+
+func (t *topnSolver) Push(x interface{}) {
+	t.rows = append(t.rows, x.(*sortRow))
+	t.heapSize++
+}
+
+func (t *topnSolver) Pop() interface{} {
+	return nil
+}
+
+func (t *topnSolver) Less(i, j int) bool {
```

Comment:
extract the dumplicated code to be a method as **less([]*tipb.ByItem)**
or
directly invoke t.topnContainer.Less() 
 may be better